todo/tests.py

```diff
@@ -1,6 +1,141 @@
+        self.assertEqual(len(response.context_data["todo"]), 1)
+        self.assertEqual(len(response.context_data["in_progress"]), 1)
+        self.assertEqual(len(response.context_data["done"]), 1)
+        with self.assertRaises(KeyError):
+            response.context_data["to_delete"]
+
+    def test_detail_view_should_display_correct_record(self):
+        request = self.f.get(reverse('taskdetail', kwargs={'pk': "1"}))
+        request.user = self.user
+        view = TaskDetailView.as_view()
+        kwargs = {'pk': "1"}
+        response = view(request, **kwargs)
+        response.render()
+        self.assertTrue(response.context_data['object'])
+        self.assertEqual(response.context_data['object'].text, "some_text_0")
+
+    def test_update_view_should_update_correct_record(self):
+        request = self.f.get(reverse('updatestatus', kwargs={"pk": "1"}))
+        request.user = self.user
+        form_data = {"status": "in_progress"}
+        request.form = TaskUpdateForm(instance=self.task0, data=form_data)
+        self.assertTrue(request.form.is_valid())
+
+        request.form.save()
+        kwargs = {"pk": "1"}
+        view = UpdateStatusView.as_view()
+        response = view(request, **kwargs)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.task0.status, "in_progress")
+
+    def test_task_form_view_should_have_correct_initial_values(self):
+        request = self.f.post(reverse('taskform', kwargs={"status": "todo"}))
+        request.user = self.user
+        form_data = {"title": "next_title", "text": "more text"}
+        request.form = TaskForm(data=form_data)
+        kwargs = {"status": "todo"}
+        view = TaskFormView.as_view()
+        response = view(request, **kwargs)
+        self.assertTrue(response.status_code, 200)
+        self.assertEqual(response.context_data['form'].initial, {"status": "todo", "author": UserFactory()})
+
+    def test_form_task_should_be_valid_and_save_record(self):
+        form_data = {"title": "next_title", "text": "more text", "status": "todo", "author": "1"}
+        form = TaskForm(data=form_data)
+        self.assertTrue(form.is_valid())
+
+        form.save()
+        self.assertEqual(Task.objects.count(), 5)
+
+    def test_delete_should_delete_selected_task(self):
+        request = self.f.post(reverse('deletetask', kwargs={"pk": "1"}))
+        request.user = self.user
+        view = DeleteTaskView.as_view()
+        kwargs = {"pk": "1"}
+        response = view(request, **kwargs)
+        self.assertEqual(response.status_code, 302)
+        with self.assertRaises(ObjectDoesNotExist):
+            Task.objects.get(pk=1)
+
+    def test_signup_view_should_redirect_to_correct_template(self):
+        request = self.f.get(reverse('signupview'))
+        view = SignupUserView.as_view()
+        response = view(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(template_name=reverse('signupview'))
```

Comment:
It doesn't work assertTemplateUsed is taking two parameters: response, and template_name, reverse is not returning template !
`self.assertTemplateUsed(response, template_name)`

Reply:
You can't use assertTemplateUsed in tests in which you didn't use Client() for tests.